### ISSUE TITLE
Validate_intergraph_edge needs updating

### ISSUE THREAD

**Issue by acl-cqc** [CQCL-DEV/hugr#231]
(See below)
1. Current spec says Static edges should be either Ext or Dom (rather than "from/to anywhere"), just that if Ext they don't need Order edges.
2. I note we check the type of Static edges from Const nodes but not from function definitions or declarations - is this correct? (This seems an odd place to check types at all, but if we are so doing, shouldn't it apply regardless of source node)

https://github.com/CQCL-DEV/hugr/blob/161adaebb2fc5c24736ef7ccb2c97a7b4bfcfd2c/src/hugr/validate.rs#L433-L451

**Comment by acl-cqc** [comment]
@ss2165 @aborgna-q any comments?

**Comment by aborgna-q** [comment]
Yeah, I agree that it needs some updating, and the typecheck could be moved somewhere else.

Also, the docstring for that method still references copy nodes.
